packages/admin-panel-server/examples.http

```diff
@@ -0,0 +1,113 @@
+@baseUrl = http://localhost:8070/v1
+# Uncomment below to fetch against the dev server, rather than localhost
+# @baseUrl = https://dev-admin-api.tupaia.org/v1
+@contentType = application/json
+# In order to setup authorization, please set 'email' and 'password' in your restClient environement variables
+# see: https://marketplace.visualstudio.com/items?itemName=humao.rest-client#environment-variables
+@authorization  = Basic {{email}}:{{password}}
+
+### Fetch Dashboard Preview Data
+POST {{baseUrl}}/fetchReportPreviewData?entityCode=TO
+content-type: {{contentType}}
+Authorization: {{authorization}}
+
+{
+    "previewConfig": {
+        "data": {
+            "dataElements": [
+                "BCD46",
+                "BCD47",
+                "BCD48",
+                "BCD48a",
+                "BCD48b",
+                "BCD49",
+                "BCD50",
+                "BCD50",
+                "BCD51",
+                "BCD52",
+                "BCD53",
+                "BCD54",
+                "BCD55"
+            ],
+            "aggregations": ["MOST_RECENT"],
+            "transform": [
+                "keyValueByDataElementName",
+                {
+                    "transform": "select",
+                    "'Doctors'": "$row.BCD46",
+                    "'Midwives'": "$row.BCD47",
+                    "'Nurses'": "sum([$row.BCD48, $row.BCD48a, $row.BCD48b])",
+                    "'Aides'": "$row.BCD49",
+                    "'Others'": "sum([$row.BCD50, $row.BCD51, $row.BCD52, $row.BCD53, $row.BCD54, $row.BCD55])",
+                    "...": ["period", "organisationUnit"]
+                },
+                {
+                    "transform": "aggregate",
+                    "organisationUnit": "group",
+                    "period": "group",
+                    "Doctors": "sum",
+                    "Midwives": "sum",
+                    "Nurses": "sum",
+                    "Aides": "sum",
+                    "Others": "sum"
+                }
+            ]
+        }
```

Comment:
I've actually noticed a bit of a tricky bit of logic we have here...
The report-server has a final 'output' step where it is able to reformat that base table that comes from the transform step into something that the frontend can consume. A few things to note about the output step:
1. We plan on having it's configuration be specified in the 'Presentation' section of the config
2. Depending on whether we are generating a 'Data' or 'Chart' preview, we conditionally want to include it in the config we send the to the report-server. If we are viewing the 'Data' table, then we don't want the report-server to process the output step as otherwise the data will come back in weird shape, but for Chart preview we want it to be processed.

I guess this brings up a few things we need to consider:
1. Who makes the call to decide if we send through to the report-server. The orchestration server here? Or does the front-end edit what's sent through to this endpoint based on which tab the user is on?
2. How do we avoid constantly refetching data whenever the user jumps between tabs? I assume by front-end caching? Should be fine 👍 
3. Who works out how to combine the 'Output' section of the Presentation config with the Report? Frontend or backend?

I'd probably lean towards backend for both 1, and 3 👍 I imagine the frontend will send through the following:
```
POST {{baseUrl}}/fetchReportPreviewData?entityCode=TO&previewMode=data
{
    "previewConfig": <all viz config>
}     
```
And then the backend works out what to send to the report-server based on which preview mode it is (data/chart). Would you agree?

Reply:
Ah yes, definitely agree! In fact I'd point the same idea out in the description of this PR (point 2):

> 2. All the endpoints in the backend that perform operations related to visualisations should receive the full payload (the whole representation mentioned above) regardless, even if some information may be redundant.
>
> The reason is most of the time, operations will need the whole payload. In some special cases like when fetching preview data, we don't need the whole payload, but still need a bit of information here and there between the data and presentation to build the report config (because report's output step actually lives in presentation instead of data). I also think sending the whole viz representation as payload will also help remove the need of the front-end to be aware of the existence of dashboard_item/report, instead, that responsibility can be delegated to the orchestration server. The front end doesn't have to do the tricky part of extracting the right information and just have to be aware of only one thing which is the visual representation.

Reply:
Oh sorry that I missed your point! You worded it perfectly, totally agree with you on that 👍